qiita_db/meta_util.py

```diff
@@ -163,3 +168,34 @@ def get_accessible_filepath_ids(user):
             filepath_ids.update(analysis.all_associated_filepath_ids)
 
         return filepath_ids
+
+
+def get_lat_longs():
+    """Retrieve the latitude and longitude of all the samples in the DB
+
+    Returns
+    -------
+    list of [float, float]
+        The latitude and longitude for each sample in the database
+    """
+    portal_table_ids = Portal(qiita_config.portal).get_studies()
+
+    with TRN:
+        sql = """SELECT DISTINCT table_name
+                 FROM information_schema.columns
+                 WHERE SUBSTR(table_name, 1, 7) = 'sample_'
+                    AND table_schema = 'qiita'
+                    AND column_name IN ('latitude', 'longitude');"""
+        TRN.add(sql)
+
+        sql = "SELECT latitude, longitude FROM qiita.{0}"
+        idx = TRN.index
+
+        portal_tables = TRN.execute_fetchflatten()
+        portal_tables = [x for x in TRN.execute_fetchflatten()
```

Comment:
This for loop can be removed if we modify the original SQL query to do the filtering for us:

```python
sql = """SELECT DISTINCT table_name
         FROM information_schema.columns
         WHERE SUBSTR(table_name, 1, 7) = 'sample_'
             AND table_name != 'sample_template_filepath'
             AND table_schema = 'qiita'
             AND column_name IN ('latitude', 'longitude');
             AND SPLIT_PART(table_name, '_', 2)::int IN %s
TRN.add(sql, [tuple(portal_table_ids)])
portal_tables = TRN.execute_fetchflatten()
```
And then do the `for` loop to add the sql for each table.